pkg/network/usm/ebpf_main.go

```diff
@@ -353,6 +361,10 @@ func (e *ebpfProgram) init(buf bytecode.AssetReader, options manager.Options) er
 		manager.ConstantEditor{Name: "ephemeral_range_begin", Value: uint64(begin)},
 		manager.ConstantEditor{Name: "ephemeral_range_end", Value: uint64(end)})
 
+	// Enable event flushes from socket filter programs if possible
+	isDirectFlushSupported := e.kversion >= kernel.VersionCode(5, 4, 0)
```

Comment:
FYI you can do more direct feature checking with `features.HaveProgramHelper(ebpf.SocketFilter, asm.FnPerfEventOutput)`

Reply:
TIL
Nice! I'll cut a follow-up PR for that